crates/abi/src/lib.rs

```diff
@@ -120,21 +120,12 @@ macro_rules! firmware_panic {
 // ————————————————————————————————— Utils —————————————————————————————————— //
 
 #[inline]
-unsafe fn miralis_ecall(fid: usize) -> Result<usize, usize> {
-    let error: usize;
-    let value: usize;
-
+pub unsafe fn miralis_end_benchmark() -> ! {
     asm!(
         "ecall",
-        in("a6") fid,
+        in("a6") 0x3,
         in("a7") abi::MIRALIS_EID,
-        out("a0") error,
-        out("a1") value,
     );
 
-    if error != 0 {
-        Err(error)
-    } else {
-        Ok(value)
-    }
+    panic!("Should not return from ending benchmarking.")
 }
```

Comment:
Sorry I might not have been super clear, we need to keep the `miralis_ecall` as it is used by other functions, but not make it public. You can create a `miralis_end_benchmark` function (and make it public) that calls into `miralis_ecall` internally.

You can put it above in the file with the other public-facing functions:

https://github.com/CharlyCst/miralis/blob/fef8730a352c367af5897810caa0c44136239570/crates/abi/src/lib.rs#L19-L40

Reply:
My bad, you were clear enough, I just forgot that.